.eslintrc.js

```diff
@@ -23,4 +23,9 @@
 
 module.exports = {
   extends: '@react-native-community',
+  env: {
+    jest: true,
```

Comment:
this is interesting - I have not had to add a jest env to my eslints before, what happens if this is not included?
The react-native template just does a react-native template does not do this https://github.com/facebook/react-native/blob/765844055ba0d02262a11114bad5da67e935d8df/template/_eslintrc.js#L3 :thinking: - I am probably just missing something as this is new to me

It appears it is pulling this plugin https://github.com/jest-community/eslint-plugin-jest#readme and that is to enforce good jest practices? 

But the plugin is already there from `@react-native-community` ? https://github.com/facebook/react-native/blob/765844055ba0d02262a11114bad5da67e935d8df/packages/eslint-config-react-native-community/index.js#L27

So I am confused :-)

Reply:
This is to avoid this error


![Screen Shot 2021-11-19 at 11 30 06](https://user-images.githubusercontent.com/11574033/142639218-e51ad71b-8c05-4fea-ac4f-337da295f7ae.png)


We need to tell eslint that we are using jest.

Reply:
I get that - I am really trying to figure out why we extend `@react-native-community` and in there *it does this*

https://github.com/facebook/react-native/blob/765844055ba0d02262a11114bad5da67e935d8df/packages/eslint-config-react-native-community/index.js#L66-L69

It should already be configured. This is very confusing as it appears we're just doing what we have done already. Why isn't `@react-native-community` working ? If the jest part is not working, is there some other part not working ?

Reply:
I'll check it @mikehardy

Reply:
@mikehardy i think i understand it, look the override from `react-native-community`:

```
files: [
  '*.{spec,test}.{js,ts,tsx}',
  '**/__{mocks,tests}__/**/*.{js,ts,tsx}',
],
env: {
   jest: true,
  'jest/globals': true,
},
rules: {
  'react-native/no-inline-styles': 0,
  quotes: [1, 'single', {avoidEscape: true, allowTemplateLiterals: true}],
},
}, 
```

they're applying override only for files that contains `spec` or `test`, that's why is not working for us, because my setup file is just a `index.js` file.

What do you think?

Reply:
perhaps reduce complexity by having it be index.test.js ? would that work - fewer moving parts / smaller changes are usually a plus

Reply:
My opinion about this is that a file with `test`  or `spec` should have a testing case, the setup file should not use one of those prefixes.

That's how I see it, what about you?

Reply:
My opinion of putting it in the env with no filter is that it defines jest globals in files that should not have them. The setup file is definitely test related. Perhaps as a middle ground, just define the jest global directly in the mock using the eslint syntax to do so ? 

> /* eslint-env jest */

This is a super super minor point though of course, I don't want the general awesomeness of having jest integrated in CI and a mock available to everyone to be overshadowed at all!

Reply:
that's an awesome idea! done 278a337c7d57618a42225246823d3403dba7cac1